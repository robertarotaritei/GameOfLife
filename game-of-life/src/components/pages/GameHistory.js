import React from 'react';
import NavigationBar from '../NavigationBar';
import HistoryList from '../History/HistoryList';

class GameHistory extends React.Component {
    constructor() {
        super();

        this.state = {
            games: []
        }
    }

    componentDidMount(){
        fetch(`/history/gamehistory`)
            .then(res => res.json())
            .then((data) => {
                this.setState({ games: data })
        })
    }

    render() {
        return (
            <div className="gameHistory">
                <NavigationBar />
                <div className="container">
                    <HistoryList games={this.state.games}/>
                </div>
            </div>
        );
    }
}

export default GameHistory;