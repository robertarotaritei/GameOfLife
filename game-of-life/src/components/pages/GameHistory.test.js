import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import GameHistory from './GameHistory';
import { shallow } from "enzyme";

test("GameHistory renders without crashing", () => {
  shallow(<GameHistory />);
});

test('Render GameHistory', () => {
  const { getByText } = render(<GameHistory />);
  const linkElement = getByText(/Game History/i);
  expect(linkElement).toBeInTheDocument();
});

it('GameHistory renders correctly', () => {
  const tree = renderer
    .create(<GameHistory />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});