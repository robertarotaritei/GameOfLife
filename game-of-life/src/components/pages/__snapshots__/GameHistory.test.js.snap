// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameHistory renders correctly 1`] = `
<div
  className="gameHistory"
>
  <div
    className="topnav"
  >
    Welcome 
    
    <button
      className="linkbutton"
      onClick={[Function]}
    >
      Dashboard
    </button>
    <button
      className="linkbutton"
      onClick={[Function]}
    >
      Game History
    </button>
    <button
      className="linkbutton"
      onClick={[Function]}
    >
      Log out
    </button>
  </div>
  <div
    className="container"
  >
    <div>
      <div
        style={
          Object {
            "paddingBottom": 20,
          }
        }
      />
    </div>
  </div>
</div>
`;
